core/comp_ui.py

```diff
@@ -85,13 +88,37 @@ def __init__(self):
 class _IDisplay(object):
     """Interface for completion displays."""
 
-    def __init__(self, comp_state, prompt_state, num_lines_cap, f, debug_f):
-        # type: (State, PromptState, int, mylib.Writer, _DebugFile) -> None
+    def __init__(self, comp_state, prompt_state, num_lines_cap, f, debug_f, signal_safe):
+        # type: (State, PromptState, int, mylib.Writer, _DebugFile, iolib.SignalSafe) -> None
         self.comp_state = comp_state
         self.prompt_state = prompt_state
         self.num_lines_cap = num_lines_cap
         self.f = f
         self.debug_f = debug_f
+        self.term_width = DEFAULT_TERM_WIDTH
```

Comment:
I think we should have a free function `_GetTermWidth()` that does the try/except, and that will be called in `__init__` here

And then we will have a **method** `self._MaybeGetTermWidth()` that does the SIGWINCH thing

that will have a bit less duplication with DEFAULT_TERM_WIDTH

Reply:
Good idea. Pulled that logic out of the method. I left the method name alone, though. "Maybe" suggests that it might fail or something and return None/-1, which I thought was more confusing than the name collision (the "self." disambiguates this enough IMO)